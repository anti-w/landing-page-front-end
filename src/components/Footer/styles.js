import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.small};
    text-align: center;
  `}
`;
