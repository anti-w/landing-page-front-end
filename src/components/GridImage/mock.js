export default {
  background: false,
  title: 'Gallery',
  description: 'The images are from unsplash 300 x 300',
  grid: [
    {
      altText: 'Title one',
      srcImg:
        'https://source.unsplash.com/random/800x800r=1',
    },
    {
      altText: 'Title two',
      srcImg:
        'https://source.unsplash.com/random/800x800r=2',
    },
    {
      altText: 'Title three',
      srcImg:
        'https://source.unsplash.com/random/800x800r=3',
    },
    {
      altText: 'Title four',
      srcImg:
        'https://source.unsplash.com/random/800x800r=4',
    },
    {
      altText: 'Title five',
      srcImg:
        'https://source.unsplash.com/random/800x800r=5',
    },
    {
      altText: 'Title six',
      srcImg:
        'https://source.unsplash.com/random/256x256r=6',
    },
  ],
};
