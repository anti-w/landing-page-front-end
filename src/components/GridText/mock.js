export default {
  background: false,
  title: 'My grid',
  description:
    'Simple description like a caption or something, resumw',
  grid: [
    {
      title: 'Title one',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum non libero tortor. Sed id nibh sed magna euismod ultrices at nec tellus. Aliquam laoreet et arcu sed convallis. Suspendisse iaculis placerat efficitur. Nunc iaculis magna eu justo ornare volutpat. Maecenas. ',
    },
    {
      title: 'Title two',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque nec nisi mauris. Ut at risus in nibh interdum imperdiet. Sed. ',
    },
    {
      title: 'Title three',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque finibus. ',
    },
  ],
};
