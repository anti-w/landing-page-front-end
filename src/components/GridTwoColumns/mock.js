export default {
  title: 'Grid Two Columns',
  text: `All the world, a lot of people! U are the finnest, I ever know. Tantz tantz tantzz`,
  srcImg: 'assets/images/joy.svg',
};
