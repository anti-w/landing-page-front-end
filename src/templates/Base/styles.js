import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 7rem;
    @media ${theme.media.lteMedium} {
      padding-top: 0;
    }
  `}
`;
