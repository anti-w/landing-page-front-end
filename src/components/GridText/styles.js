import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextContainer } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextContainer} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(
      auto-fill,
      minmax(280px, 1fr)
    );
    gap: ${theme.spacings.large};
    width: 100%;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
      margin-bottom: 2rem;
    }
    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 6rem;
      top: -2rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
