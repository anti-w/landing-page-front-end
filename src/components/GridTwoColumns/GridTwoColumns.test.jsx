import { screen } from '@testing-library/react';

import { GridTwoColumns } from '.';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<GridTwoColumns />', () => {
  it('should match a snapshot of a grid with two columns', () => {
    const { container } = renderTheme(
      <GridTwoColumns {...mock} />,
    );
    expect(container).toMatchSnapshot();
  });
});
