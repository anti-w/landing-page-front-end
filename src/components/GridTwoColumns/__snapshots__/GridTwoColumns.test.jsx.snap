// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should match a snapshot of a grid with two columns 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 3.2rem;
}

.c7 {
  max-width: 72vh;
  max-height: 72vh;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c7 {
    max-width: 55vh;
    max-height: 55vh;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
          >
            Grid Two Columns
          </h1>
          <div
            class="c6"
          >
            All the world, a lot of people! U are the finnest, I ever know. Tantz tantz tantzz
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid Two Columns"
            class="c7"
            src="assets/images/joy.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
